bees/httpbee/httpbee.go

```diff
@@ -86,16 +119,39 @@ func (mod *HTTPBee) Action(action bees.Action) []bees.Placeholder {
 		var j string
 		var form url.Values
 		var resp *http.Response
+		var h []string
+		var ctype string
 
 		action.Options.Bind("json", &j)
 		action.Options.Bind("form", &form)
+		action.Options.Bind("headers", &h)
+
+		var buf *strings.Reader
 
 		if j != "" {
-			buf := strings.NewReader(j)
-			resp, err = http.Post(u, "application/json", buf)
+			buf = strings.NewReader(j)
+			ctype = "application/json"
 		} else {
-			resp, err = http.PostForm(u, form)
+			buf = strings.NewReader(form.Encode())
+			ctype = "application/x-www-form-urlencoded"
+		}
+
+		req, err := http.NewRequest("POST", u, buf)
+		req.Header.Set("Content-Type", ctype)
+
+		if err != nil {
+			mod.LogErrorf("Error: %s", err)
+			return outs
+		}
+		for _, header := range h {
```

Comment:
I feel like there's a bit of code duplication going on here. We could probably move the header-parsing into a separate little helper method.

Reply:
Whoops! Some of the duplication here is a bit obvious, huh...

Should all be fixed with this next commit 😄